subscription.go

```diff
@@ -0,0 +1,219 @@
+package opcua
+
+import (
+	"context"
+	"math"
+	"time"
+
+	"github.com/gopcua/opcua/ua"
+)
+
+const (
+	DefaultSubscriptionMaxNotificationsPerPublish = 10000
+	DefaultSubscriptionLifetimeCount              = 10000
+	DefaultSubscriptionMaxKeepAliveCount          = 3000
+	DefaultSubscriptionInterval                   = 100 * time.Millisecond
+	DefaultSubscriptionPriority                   = 0
+)
+
+type Subscription struct {
+	SubscriptionID            uint32
+	RevisedPublishingInterval float64
+	RevisedLifetimeCount      uint32
+	RevisedMaxKeepAliveCount  uint32
+	Notifs                    chan *PublishNotificationData
+	c                         *Client
+}
+
+type SubscriptionParameters struct {
+	Interval                   time.Duration
+	LifetimeCount              uint32
+	MaxKeepAliveCount          uint32
+	MaxNotificationsPerPublish uint32
+	Priority                   uint8
+	Notifs                     chan *PublishNotificationData
+}
+
+func NewMonitoredItemCreateRequestWithDefaults(nodeID *ua.NodeID, attributeID ua.AttributeID, clientHandle uint32) *ua.MonitoredItemCreateRequest {
+	if attributeID == 0 {
+		attributeID = ua.AttributeIDValue
+	}
+	return &ua.MonitoredItemCreateRequest{
+		ItemToMonitor: &ua.ReadValueID{
+			NodeID:       nodeID,
+			AttributeID:  attributeID,
+			DataEncoding: &ua.QualifiedName{},
+		},
+		MonitoringMode: ua.MonitoringModeReporting,
+		RequestedParameters: &ua.MonitoringParameters{
+			ClientHandle:     clientHandle,
+			DiscardOldest:    true,
+			Filter:           nil,
+			QueueSize:        10,
+			SamplingInterval: 0.0,
+		},
+	}
+}
+
+type PublishNotificationData struct {
+	SubscriptionID uint32
+	Error          error
+	Value          interface{}
+}
+
+// Cancel() deletes the Subscription from Server and makes the Client forget it so that publishing
```

Comment:
Can we rename this to `Close()` to be more in line with the stdlib?

Reply:
I have no strong opinion on this, but it really seems to me that the natural thing to do to stop a subscription is to "cancel" it. Just like you don't "repeat" a subscription, but rather "renew" it.

Reply:
How about introducing `Close()` as a synonym to `Cancel()`? E.g.
```go
func (s *Subscription) Close() error {
    return s.Cancel()
}
```

Reply:
Either Close or Cancel. I’ll do some digging but this isn’t that important.